pymc_marketing/mmm/mmm.py

```diff
@@ -2239,28 +2239,14 @@ def plot_budget_allocation(
             The matplotlib figure object and axis containing the plot.
 
         """
-        if original_scale:
-            channel_contributions = (
-                samples["channel_contributions"]
-                .mean(dim=["sample"])
-                .mean(dim=["date"])
-                .values
-                * self.get_target_transformer()["scaler"].scale_
-            )
+        channel_contributions = (
+            samples["channel_contributions"].mean(dim=["date", "sample"]).to_numpy()
+        )
 
-            allocate_spend = (
-                np.array(list(self.optimal_allocation_dict.values()))
-                * self.channel_transformer["scaler"].scale_
-            )
+        if original_scale:
+            channel_contributions *= self.get_target_transformer()["scaler"].scale_
```

Comment:
@carlosagostini, Is this still valid, or should we remove the scaling everywhere?

Reply:
Contributions still need to be scaled, but not the budget allocation, because the optimizer now returns it in original scale. So post processing is not necessary 🙌🏻

Reply:
Ok! Thanks! Si is this one good to go ?

Reply:
Indeed!